### PR TITLE
Yuanqing/feature trecord

### PR DESCRIPTION
**Pull request by Yqzzxlj**
Added serializers.py, views.py and tests_serializers.py, test__views.py
Added training_record/urls.py
Added url in TMSFTT/urls.py for training_record

### CODE REVIEW

**Review by ghh82058 — DISMISSED** [review]

**Review by a8520238 — DISMISSED** [review]

File: TMSFTT/urls.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 
 API_URLPATTERNS = [
+    path('training_record', include('training_record.urls')),
```
Comment (Time1ess) [inline TMSFTT/urls.py]:
```suggestion
    path('training-record', include('training_record.urls')),
```
As suggested in https://github.com/DLUT-SIE/TMSFTT-BE/pull/9/files#r248498944

File: training_record/serializers.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+import training_record.models
+
+
+User = get_user_model()
+
+
+class UserSerializer(serializers.ModelSerializer):
```
Comment (Time1ess) [inline training_record/serializers.py]: Delete this.
As comment at https://github.com/DLUT-SIE/TMSFTT-BE/pull/10/files#r248509481

File: training_record/tests/tests_serializers.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+from django.test import TestCase
+
+import training_record.serializers as serializers
+import training_record.models as models
+
+
+User = get_user_model()
+
+
+class TestUserSerializer(TestCase):
```
Comment (Time1ess) [inline training_record/tests/tests_serializers.py]: Delete this.

File: training_record/tests/tests_serializers.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+        keys = set(serializers.RecordSerializer(record).data.keys())
+        self.assertEqual(keys, expected_keys)
+
+
+class TestRecordContentSerializer(TestCase):
+    '''Unit tests for serializer of RecordContent.'''
+    def test_fields_equal(self):
+        '''Serializer should return fields of RecordContent correctly.'''
+        recordcontent = models.RecordContent()
```
Comment (Time1ess) [inline training_record/tests/tests_serializers.py]:
```suggestion
        record_content = models.RecordContent()
```

File: training_record/tests/tests_serializers.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        keys = set(serializers.RecordContentSerializer(recordcontent).
+                   data.keys())
+        self.assertEqual(keys, expected_keys)
+
+
+class TestRecordAttachmentSerializer(TestCase):
+    '''Unit tests for serializer of RecordAttachment.'''
+    def test_fields_equal(self):
+        '''Serializer should return fields of RecordAttachment correctly.'''
+        recordattachment = models.RecordAttachment()
```
Comment (Time1ess) [inline training_record/tests/tests_serializers.py]: Ditto.

File: training_record/tests/tests_serializers.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        keys = set(serializers.RecordAttachmentSerializer(recordattachment).
+                   data.keys())
+        self.assertEqual(keys, expected_keys)
+
+
+class StatusChangeLogSerializer(TestCase):
+    '''Unit tests for serializer of StatusChangeLog.'''
+    def test_fields_equal(self):
+        '''Serializer should return fields of StatusChangeLog correctly.'''
+        statuschangelog = models.StatusChangeLog()
```
Comment (Time1ess) [inline training_record/tests/tests_serializers.py]: Ditto.

File: training_record/tests/tests_views.py [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+        user = mommy.make(User)
+        url = reverse('record-list')
+        data = {'campus_event': campus_event.id,
+                'user': user.id}
+
+        response = self.client.post(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(training_record.models.Record.objects.count(), 1)
+        self.assertEqual(training_record.models.Record.objects.get().
```
Comment (Time1ess) [inline training_record/tests/tests_views.py]: https://github.com/DLUT-SIE/TMSFTT-BE/pull/9/files#r248504817.

File: training_record/tests/tests_views.py [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+        response = self.client.patch(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn('status', response.data)
+        self.assertEqual(response.data['status'], status1)
+
+
+class TestRecordContentViewSet(APITestCase):
+    '''Unit tests for RecordContent view.'''
+    def test_create_recordcontent(self):
```
Comment (Time1ess) [inline training_record/tests/tests_views.py]:
```suggestion
    def test_create_record_content(self):
```

File: training_record/tests/tests_views.py [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+        self.assertEqual(training_record.models.RecordContent.objects.count(),
+                         1)
+        self.assertEqual(training_record.models.RecordContent.objects.get().
+                         record.id, record.id)
+        self.assertEqual(training_record.models.RecordContent.objects.get().
+                         content_type, content_type)
+        self.assertEqual(training_record.models.RecordContent.objects.get().
+                         content, content)
+
+    def test_list_recordcontent(self):
```
Comment (Time1ess) [inline training_record/tests/tests_views.py]:
```suggestion
    def test_list_record_content(self):
```

File: training_record/tests/tests_views.py [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+        '''RecordAttachment should be created by POST request.'''
+        url = reverse('recordattachment-list')
+        campus_event = mommy.make(training_event.models.CampusEvent)
+        record = mommy.make(training_record.models.Record,
+                            campus_event=campus_event)
+        attachment_type = 0
+        path = io.BytesIO(b'some content')
+        data = {'record': record.id, 'attachment_type': attachment_type,
+                'path': path}
+        response = self.client.post(url, data, format='multipart')
```
Comment (Time1ess) [inline training_record/tests/tests_views.py:190]: https://github.com/DLUT-SIE/TMSFTT-BE/pull/10/files#r248511133

File: training_record/tests/tests_views.py [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+
+class TestRecordAttachmentViewSet(APITestCase):
+    '''Unit tests for RecordAttachment view.'''
+    def test_create_recordattachment(self):
+        '''RecordAttachment should be created by POST request.'''
+        url = reverse('recordattachment-list')
+        campus_event = mommy.make(training_event.models.CampusEvent)
+        record = mommy.make(training_record.models.Record,
+                            campus_event=campus_event)
+        attachment_type = 0
```
Comment (Time1ess) [inline training_record/tests/tests_views.py]: Do not use 0 directly, you can use `RecordAttachment.ATTACHMENT_TYPE_CONTENT` to replace it.

**Review by Time1ess — CHANGES_REQUESTED** [review]

File: TMSFTT/urls.py [hunk 12]
```diff
@@ -19,6 +19,7 @@
 
 
 API_URLPATTERNS = [
+    path('training-record', include('training_record.urls')),
```
Comment (Time1ess) [inline TMSFTT/urls.py]: While this is OK, it's better to keep this close to other training* routes, so please move it down.

**Review by Time1ess — APPROVED** [review]

**Review by a8520238 — APPROVED** [review]

**Pull request merged**